pkg/workspace_repo/workspace_repo.go

```diff
@@ -16,19 +16,20 @@ type Workspace interface {
 	WorkspaceName() string
 	GetCurrentRunID() string
 	GetLatestConfigVersionID() (string, error)
-	Run(isDestroy bool) error
+	Run(isDestroy bool, dryRun bool) error
```

Comment:
I know this would require a large refactor of what you have, but since we already have this interface, I wonder if instead of adding the dryRun bool to these methods, make a separate implementation of Workspace that is called something like `WorkspaceDryRunner` and when the dry-run flag is set in the command, `GetWorkspace()` returns the `WorkspaceDryRunner` that handles all the implementation details of running these functions in dry-run mode. Some psedocode:

~~~go
creatorWorkspace, err := s.workspaceRepo.GetWorkspace(ctx, s.creatorWorkspaceName, dryRun)

////


func (c *WorkspaceRepo) GetWorkspace(ctx context.Context, workspaceName string, dryRun bool) (Workspace, error) {
  if dryRun {
return WorkspaceDryRunner
}
return regularWorkspace
}
~~~

Reply:
This is where the fundamental issue is. The difference between the dry run and a regular run is a `speculative` flag on the configuration, rest of the code in workspace.go is the same. I feel like creating a separate implementation will just duplicate the code, `speculative` is the only parameter that varies between the two.

Reply:
That's fair! Ignore my comments then